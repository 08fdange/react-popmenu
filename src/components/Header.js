import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const propTypes = {
  title: PropTypes.string,
  children: PropTypes.any,
};

const Wrapper = styled.div`
  position: fixed;
  z-index: 999;
  top: 0;
  height: 78px;
  width: calc(100% - 32px);
  background-color: white;
  border-bottom: 1px solid #e6e6e6;
  box-shadow: rgb(166 179 194 / 30%) 0px 4px 14px 0px;
  padding: 0 16px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const Title = styled.h1`
  margin: 0;
  padding: 0;
`;

const Header = (props) => {
  const { title, children } = props;

  return(
    <Wrapper>
      <Title>{title}</Title>
      {children}
    </Wrapper>
  )
}

Header.propTypes = propTypes;

export default Header;