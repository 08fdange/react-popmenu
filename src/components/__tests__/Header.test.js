import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Header from '../Header';
import Colors from '../../theme/colors';

test('Header - renders correctly', () => {
  const tree = renderer.create(<Header title='Header' />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header - renders props correctly', () => {
  const testRenderer = renderer.create(<Header title='Header'>Test</Header>);
  const tree = testRenderer.toJSON();
  const testInstance = testRenderer.root;
  expect(testInstance.props.children).toBe('Test');
  expect(tree).toHaveStyleRule(Colors.White);
});