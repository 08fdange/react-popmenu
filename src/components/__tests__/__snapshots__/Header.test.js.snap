// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header - renders correctly 1`] = `
.c0 {
  position: fixed;
  z-index: 999;
  top: 0;
  height: 78px;
  width: calc(100% - 32px);
  background-color: white;
  border-bottom: 1px solid #e6e6e6;
  box-shadow: rgb(166 179 194 / 30%) 0px 4px 14px 0px;
  padding: 0 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  margin: 0;
  padding: 0;
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    Header
  </h1>
</div>
`;
